### ISSUE TITLE
SPSearchTopology documentation improvements

### ISSUE THREAD

**Issue by shurick81** [PowerShell/SharePointDsc#546]
Would it be great to mention on https://github.com/PowerShell/SharePointDsc/wiki/SPSearchTopology what servers should we run this configuration on and is there a necessity of provisioning search services before topology is applied?

For example, I've got s1, s2 and s3 servers in the farm and only s2 and s3 should host search components.
Should my configuration be like this?

```
Node "s2","s3"
{
        SPServiceInstance SearchHostControllerService
        {  
            Name           = "Search Host Controller Service"
            Ensure         = "Present"
            InstallAccount = $SetupAccount
        }
        SPServiceInstance SearchQueryAndSiteSettingsService
        {  
            Name           = "Search Query and Site Settings Service"
            Ensure         = "Present"
            InstallAccount = $SetupAccount
        }
        SPServiceInstance SharePointServerSearch
        {  
            Name           = "SharePoint Server Search"
            Ensure         = "Present"
            InstallAccount = $SetupAccount
        }
}
Node "s1","s2"
{
        SPSearchTopology "LocalSearchTopology"
        {
            ServiceAppName          = $serviceAppName
            Admin                   = "s1"
            Crawler                 = "s2"
            ContentProcessing       = "s2"
            AnalyticsProcessing     = "s2"
            QueryProcessing         = "s1"
            PsDscRunAsCredential    = $SetupAccount
            FirstPartitionDirectory = $searchIndexDirectory
            IndexPartition          = "s1"
        }
}

```

Or must it be like this, without instances provisioning?
```
Node "s1","s2"
{
        SPSearchTopology "LocalSearchTopology"
        {
            ServiceAppName          = $serviceAppName
            Admin                   = "s1"
            Crawler                 = "s2"
            ContentProcessing       = "s2"
            AnalyticsProcessing     = "s2"
            QueryProcessing         = "s1"
            PsDscRunAsCredential    = $SetupAccount
            FirstPartitionDirectory = $searchIndexDirectory
            IndexPartition          = "s1"
        }
}

```

**Comment by BrianFarnhill** [comment]
Hi @shurick81 - your right we need to clear this up. At this point you only need to run the topology resource on a single server in the farm, so either s1 or s2 in the example above. It will turn the components on to each server from the one place. I'll get the documentation to be updated to better reflect this.

**Comment by shurick81** [comment]
I believe SPServiceInstance that I mentioned should be run. At least I haven't managed to apply components on the servers before service instance was up and running

**Comment by BrianFarnhill** [comment]
@shurick81 You don't need to run SPServiceIsntance as the SPSearchTopology will make them start to apply the topology, but its a good idea to run it so that the services will be started later on if they are ever found to be stopped.

**Comment by ykuijs** [comment]
Updated in my next PR

**Comment by ykuijs** [comment]
Fixed in PR #680
